### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by VodichY**
task3 - https://github.com/EPAM-JS-Competency-center/cloud-development-course-initial/blob/main/task3-product-magamanent-api/task.md

1.  poiproduct-service serverless config contains configuration for 2 lambda functions, API is not working at all, but YAML configuration is correct. 
2.  The getProductsList OR getProductsById lambda function returns a correct response (POINT1) 
3.  The getProductsById AND getProductsList lambda functions return a correct response code (POINT2) 
4.  Your own Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. AND POINT1 and POINT2 are done. 

did two endpoint, but while didn't integration with fe:
ednpoint products: https://mo9jqirxsk.execute-api.eu-west-1.amazonaws.com/dev/products
endpoint product by id: https://mo9jqirxsk.execute-api.eu-west-1.amazonaws.com/dev/products/{productId}

automated configured s3/cloudfront:
s3 link:    http://vodzichs-second-app-1.s3-website-eu-west-1.amazonaws.com/
cloudfront link:  https://d13ct5iq81ntcd.cloudfront.net/

**Additional (optional) tasks (but nice to have):**

+1 - Async/await is used in lambda functions
+1 (All languages) - Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered)

score - 4

deadline 30.08.2021 23:59:59 (GMT+3)

### CODE REVIEW

**Comment by Frol8612** [comment]
It was necessary to create a product-service folder and it should have one serverless config with two lambda functions.
For example:
```
functions:
  getProductsList:
    handler: handler.getProductsList
    ...
  getProductById:
    handler: handler.getProductById
    ...
```
And handler.js export two functions.
For example:
```
module.exports.getProductsList = async (event) => { ... };
module.exports.getProductById = async (event) => { ... };
```

**Comment by VodichY** [comment]
> It was necessary to create a product-service folder and it should have one serverless config with two lambda functions.
> For example:
> 
> ```
> functions:
>   getProductsList:
>     handler: handler.getProductsList
>     ...
>   getProductById:
>     handler: handler.getProductById
>     ...
> ```
> 
> And handler.js export two functions.
> For example:
> 
> ```
> module.exports.getProductsList = async (event) => { ... };
> module.exports.getProductById = async (event) => { ... };
> ```

You right, I fixed this problem,  thanks for your comment.

**Comment by Frol8612** [comment]
Very well

File: product-service/__tests__/handler.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const {
+  getProductsList,
+  getProductById
+} = require('../handler');
+
+test('getProductsList', () => {
+  return getProductsList().then(data => {
+    expect(data.statusCode).toBe(200);
+    expect(Array.isArray(JSON.parse(data.body))).toBe(true);
```
Comment (Frol8612) [inline product-service/__tests__/handler.test.js]:
```suggestion
    expect(Array.isArray(JSON.parse(data.body))).toBeTruthy();
```

But better, 

```suggestion
    expect(JSON.parse(data.body)).toEqual(products);
```

**Pull request merged**